frontend/src/components/MapView/DateSelector/TimelineItems/index.tsx

```diff
@@ -1,31 +1,41 @@
-import React, { memo, useCallback, useMemo } from 'react';
 import {
-  createStyles,
   Fade,
   Grid,
   Tooltip,
   Typography,
   WithStyles,
+  createStyles,
   withStyles,
 } from '@material-ui/core';
 import { CreateCSSProperties } from '@material-ui/styles';
-import { merge, compact } from 'lodash';
-import { DateRangeType } from '../../../../config/types';
-import { TIMELINE_ITEM_WIDTH, formatDate } from '../utils';
+import { compact, merge } from 'lodash';
+import React, { memo, useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { DateItem, DateRangeType } from '../../../../config/types';
+import { availableDatesSelector } from '../../../../context/serverStateSlice';
 import { moment, useSafeTranslation } from '../../../../i18n';
 import { MONTH_YEAR_DATE_FORMAT } from '../../../../utils/name-utils';
+import {
+  DateCompatibleLayer,
+  getPossibleDatesForLayer,
+} from '../../../../utils/server-utils';
+import {
+  DateCompatibleLayerWithDateItems,
+  TIMELINE_ITEM_WIDTH,
+} from '../utils';
 import TooltipItem from './TooltipItem';
 
 const TimelineItems = memo(
   ({
     classes,
     intersectionDates,
     dateRange,
-    selectedLayerDates,
-    selectedLayerTitles,
     clickDate,
     locale,
+    selectedLayers,
   }: TimelineItemsProps) => {
+    const serverAvailableDates = useSelector(availableDatesSelector);
```

Comment:
Why do we need the `serverAvailableDates` here? Inside the `selectedLayers` there is a property `dateItems`.

Reply:
Yes indeed !